tensorboard/webapp/runs/store/runs_types.ts

```diff
@@ -52,6 +52,8 @@ export interface RunsDataRoutefulState {
   groupKeyToColorString: Map<string, string>;
   userSetGroupBy?: GroupBy;
```

Comment:
You may want to change this to `GroupByKey` so there is no redundant information in the `colorGroupRegexString`.

Reply:
Agree. Yet this changes includes `groupRuns` method changes and a bunch of utils(group by regex) and tests changes. Would like to separate them into another pr.

Reply:
Another PR sounds good but do note that we don't want to change API of utils, actions, and selectors.